docs/modules/ROOT/pages/guides/contract_migration.adoc

```diff
@@ -0,0 +1,39 @@
+:source-highlighter: highlight.js
+:highlightjs-languages: rust
+:github-icon: pass:[<svg class="icon"><use href="#github-icon"/></svg>]
+
+= Migrate a Contract from EVM chain
+
+This EVM template is powered by Frontier. It allows you to deploy almost any Solidity contract you have following these steps.
+
+== Step 1: Prepare and run the chain
+
+This step is basic, you can use our xref:index.adoc[guide for generic template]. It is recommended to use xref:guides/async_backing.adoc[async backing] for better experience.
+The only additional thing that you will need to specify is the EVM Chain Id in your chainspec.
+
+== Step 2: Set up the development environment
+
+For Solidity development you can use link:https://book.getfoundry.sh/getting-started/installation[Foundry]. Feel free to use any other toolkit like ethers.js or web3.py.
+
+With the chain running, to start the development you need only two values: a private key from the account with some tokens on it and a url from EVM RPC.
+
+* Private Key. If you have formed your own chainspec, you should have some keys ready. Alternatively, you can generate the key using any wallet of your choice (e.g. Metamask) and get some tokens to it. By default, we have these keys for prefunded accounts in the chainspec:
```

Comment:
I think it will be better to do either this:

```suggestion
=== Private Key 

If you have formed your own chainspec, you should have some keys ready. Alternatively, you can generate the key using any wallet of your choice (e.g. Metamask) and get some tokens to it. By default, we have these keys for prefunded accounts in the chainspec:
```

or:
```suggestion
* Private Key: If you have formed your own chainspec, you should have some keys ready. Alternatively, you can generate the key using any wallet of your choice (e.g. Metamask) and get some tokens to it. By default, we have these keys for prefunded accounts in the chainspec:
```